osu.Game.Tests/NonVisual/CustomDataDirectoryTest.cs

```diff
@@ -151,13 +152,13 @@ public void TestMigration()
                     Assert.That(!host.Storage.ExistsDirectory(Path.Combine("test-nested", "cache")));
                     Assert.That(storage.ExistsDirectory(Path.Combine("test-nested", "cache")));
 
-                    foreach (var file in OsuStorage.IGNORE_FILES)
+                    foreach (var file in osuStorage?.IgnoreFiles ?? Array.Empty<string>())
                     {
                         Assert.That(host.Storage.Exists(file), Is.True);
                         Assert.That(storage.Exists(file), Is.False);
                     }
 
-                    foreach (var dir in OsuStorage.IGNORE_DIRECTORIES)
+                    foreach (var dir in osuStorage?.IgnoreDirectories ?? Array.Empty<string>())
```

Comment:
Not sure if this is the right way to prevent a Nullref exception and it also looks weird to loop over an empty array

Reply:
You're correct in thinking this is not good code. just do a null check *before* iterating instead (you can use it to surround both the two `foreach` statements).